processor/deltatorateprocessor/processor.go

```diff
@@ -74,7 +74,14 @@ func (dtrp *deltaToRateProcessor) processMetrics(_ context.Context, md pdata.Met
 					fromDataPoint.CopyTo(newDp)
 
 					durationNanos := time.Duration(fromDataPoint.Timestamp() - fromDataPoint.StartTimestamp())
-					rate := calculateRate(fromDataPoint.DoubleVal(), durationNanos)
+					var rate float64
+					switch fromDataPoint.Type() {
+					case pdata.MetricValueTypeDouble:
+						rate = calculateRate(fromDataPoint.DoubleVal(), durationNanos)
+					case pdata.MetricValueTypeInt:
+						rate = calculateRate(float64(fromDataPoint.IntVal()), durationNanos)
+					default:
```

Comment:
The default case should return an error (unrecoverable, so that the retry mechanism won't kick in).

https://github.com/open-telemetry/opentelemetry-collector/blob/51fcb65f02055638f223c6abd3d6d2d644c696db/consumer/consumererror/permanent.go#L25-L29

Reply:
Got it